examples/passport-oauth2-login/src/authentication-strategies/oauth2-strategy.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: example-passport-oauth2-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  asAuthStrategy,
+  AuthenticationStrategy,
+  UserProfileFactory,
+} from '@loopback/authentication';
+import {StrategyAdapter} from '@loopback/authentication-passport';
+import {Profile} from 'passport';
+import {Strategy, StrategyOption} from 'passport-facebook';
+import {Strategy as GoogleStrategy, StrategyOptions as GoogleOptions} from 'passport-google-oauth2';
+import {bind, inject} from '@loopback/context';
+import {Request, HttpErrors} from '@loopback/rest';
+import {UserProfile, securityId} from '@loopback/security';
+import {UserService, UserServiceBindings, UserWithToken} from '../services';
+let oauth2Providers = require('../../oauth2-providers');
+
+
+const facebookOptions: StrategyOption = {
+  clientID: oauth2Providers['facebook-login'].clientID,
+  clientSecret: oauth2Providers['facebook-login'].clientSecret,
+  callbackURL: oauth2Providers['facebook-login'].callbackURL,
+  profileFields: oauth2Providers['facebook-login'].profileFields
+};
+
+const googleOptions: GoogleOptions = {
+  clientID: oauth2Providers['google-login'].clientID,
+  clientSecret: oauth2Providers['google-login'].clientSecret,
+  callbackURL: oauth2Providers['google-login'].callbackURL,
+  scope: ['email']
+};
```

Comment:
These options can be possibly refactored into injectable configurations.

Reply:
agreed, I am injecting into the request context in sequence.ts